### PR TITLE
[Java][Client Default] Support of AWS Signature V4

### PR DESCRIPTION
**Pull request by outscale-mdr**
Hello,
This PR is for adding the support of AWS Signature V4 into Java client for okhttp-gson for now. (see also https://github.com/OpenAPITools/openapi-generator/pull/11690) 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by outscale-mdr** [comment]
CC 
- Java: @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)
- Admin: @wing328

**Comment by outscale-mdr** [comment]
Hello,

Is somebody available for a review on this PR please ?

Thank you

File: docs/generators/java.md [hunk 1]
```diff
@@ -91,6 +91,7 @@ These options may be applied as additional-properties (cli) or configOptions (pl
 |useRxJava3|Whether to use the RxJava3 adapter with the retrofit2 library. IMPORTANT: This option has been deprecated.| |false|
 |useSingleRequestParameter|Setting this property to true will generate functions with a single argument containing all API endpoint parameters instead of one argument per parameter. ONLY jersey2, jersey3, okhttp-gson support this option.| |false|
 |webclientBlockingOperations|Making all WebClient operations blocking(sync). Note that if on operation 'x-webclient-blocking: false' then such operation won't be sync| |false|
+|withAWSV4Signature|whether to include AWS v4 signature support| |false|
```
Comment (wing328) [inline docs/generators/java.md:94]: @outscale-mdr tested locally and the result is good.

can you update the description to indicate this option only works with `okhttp-gson` library only? then I'll merge.

Comment (outscale-mdr) [reply docs/generators/java.md]: Thank you @wing328  for your time !

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java [hunk 2]
```diff
@@ -200,6 +201,7 @@ public JavaClientCodegen() {
         cliOptions.add(CliOption.newBoolean(USE_ABSTRACTION_FOR_FILES, "Use alternative types instead of java.io.File to allow passing bytes without a file on disk. Available on resttemplate, webclient, libraries"));
         cliOptions.add(CliOption.newBoolean(DYNAMIC_OPERATIONS, "Generate operations dynamically at runtime from an OAS", this.dynamicOperations));
         cliOptions.add(CliOption.newBoolean(SUPPORT_STREAMING, "Support streaming endpoint (beta)", this.supportStreaming));
+        cliOptions.add(CliOption.newBoolean(CodegenConstants.WITH_AWSV4_SIGNATURE_COMMENT, CodegenConstants.WITH_AWSV4_SIGNATURE_COMMENT_DESC, this.withAWSV4Signature));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java:204]: I wasn't clear. You will need to update this line with something like:

`CodegenConstants.WITH_AWSV4_SIGNATURE_COMMENT_DESC + " (only available for okhttp-gson library)"`

Comment (outscale-mdr) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaClientCodegen.java]: My bad, I did not execute again the generation ...

**Pull request merged**